raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -926,29 +973,16 @@ private boolean handleVoteResponse(
             if (quorum.isLeader()) {
                 logger.debug("Ignoring vote response {} since we already became leader for epoch {}",
                     partitionResponse, quorum.epoch());
-            } else if (quorum.isCandidate()) {
-                CandidateState state = quorum.candidateStateOrThrow();
-                if (partitionResponse.voteGranted()) {
-                    state.recordGrantedVote(remoteNodeId);
-                    maybeTransitionToLeader(state, currentTimeMs);
-                } else {
-                    state.recordRejectedVote(remoteNodeId);
-
-                    // If our vote is rejected, we go immediately to the random backoff. This
-                    // ensures that we are not stuck waiting for the election timeout when the
-                    // vote has become gridlocked.
-                    if (state.isVoteRejected() && !state.isBackingOff()) {
-                        logger.info("Insufficient remaining votes to become leader (rejected by {}). " +
-                            "We will backoff before retrying election again", state.rejectingVoters());
-
-                        state.startBackingOff(
-                            currentTimeMs,
-                            binaryExponentialElectionBackoffMs(state.retries())
-                        );
-                    }
-                }
+            } else if (quorum.isVotingState()) {
+                NomineeState state = quorum.votingStateOrThrow();
```

Comment:
Let's use the same word you used in the `NomineeState`: `isNomineeState()` and `nomineeStateOrThrow()`.